### PR TITLE
more support for style

### PR DESCRIPTION
**Pull request by futuremarc**
addresses #954

### CODE REVIEW

**Comment by futuremarc** [comment]
does not update the docs however

File: lib/addons/p5.dom.js [hunk 1]
```diff
@@ -912,19 +912,72 @@
...
   p5.Element.prototype.style = function(prop, val) {
+    var self = this;
+
     if (typeof val === 'undefined') {
-      var attrs = prop.split(';');
-      for (var i=0; i<attrs.length; i++) {
-        var parts = attrs[i].split(':');
-        if (parts[0] && parts[1]) {
-          this.elt.style[parts[0].trim()] = parts[1].trim();
+      if (!prop.match(/:/)) {
```
Comment (lmccart) [inline lib/addons/p5.dom.js]: let's change this to:
if (prop.indexOf(':') === -1) {

it's a little faster

File: lib/addons/p5.dom.js [hunk 2]
```diff
@@ -912,19 +912,72 @@
...
+            this.elt.style[parts[0].trim()] = parts[1].trim();
+          }
         }
       }
     } else {
-      this.elt.style[prop] = val;
-      if (prop === 'width' || prop === 'height' || prop === 'left' || prop === 'top'){
-        var numVal = val.replace(/\D+/g,'');
-        this[prop] = parseInt(numVal);
+      if (prop.match(/rotate/)) {
```
Comment (lmccart) [inline lib/addons/p5.dom.js]: let's change this to:
if (prop === 'rotate') {

File: lib/addons/p5.dom.js [hunk 3]
```diff
@@ -912,19 +912,72 @@
...
+          this.elt.style.transform += style;
+        } else if (arguments.length === 4) {
+          var style = this.elt.style.transform.replace(/rotate3d\(.*\)/g, '');
+          style = style.replace(/rotate[X-Z]?\(.*\)/g, '');
+          this.elt.style.transform = 'rotateX(' + arguments[0] + 'deg)';
+          this.elt.style.transform += 'rotateY(' + arguments[1] + 'deg)';
+          this.elt.style.transform += 'rotateZ(' + arguments[2] + 'deg)';
+          this.elt.style.transform += style;
+        }
+      } else if (prop.match(/translate/)) {
```
Comment (lmccart) [inline lib/addons/p5.dom.js]: let's change this to:
if (prop === 'translate') {

File: lib/addons/p5.dom.js [hunk 4]
```diff
@@ -912,19 +912,72 @@
...
+          this.elt.style.transform += style;
+          this.elt.parentElement.style.perspective = '1000px';
+        } else if (arguments.length === 5) {
+          var style = this.elt.style.transform.replace(/translate3d\(.*\)/g, '');
+          style = style.replace(/translate[X-Z]?\(.*\)/g, '');
+          this.elt.style.transform = 'translate3d(' + arguments[0] + 'px,' + arguments[1] + 'px,' + arguments[2] + 'px)';
+          this.elt.style.transform += style;
+          this.elt.parentElement.style.perspective = arguments[3] + 'px';
+        }
+      } else if (prop.match(/position/)) {
```
Comment (lmccart) [inline lib/addons/p5.dom.js]: let's change this to:
if (prop === 'position') {

File: lib/addons/p5.dom.js [hunk 5]
```diff
@@ -912,19 +912,72 @@
...
+        this.elt.style.left = arguments[1] + 'px';
+        this.elt.style.top = arguments[2] + 'px';
+        this.x = arguments[1];
+        this.y = arguments[2];
+      } else {
+        this.elt.style[prop] = val;
+        if (prop === 'width' || prop === 'height' || prop === 'left' || prop === 'top') {
+          var numVal = val.replace(/\D+/g, '');
+          this[prop] = parseInt(numVal);
```
Comment (lmccart) [inline lib/addons/p5.dom.js]: parseInt(numVal, 10); 
(specify second parameter to guarantee consistent behavior across browsers)

**Comment by lmccart** [comment]
hey @futuremarc this looks great, I just left a few small notes. do you mind updating the documentation before we merge this in? this way we ensure the actual functionality doesn't diverge from the documented functionality.

**Comment by futuremarc** [comment]
sure!

**Comment by lmccart** [comment]
:)

**Pull request merged**